### PR TITLE
Update generating-a-json-web-token-jwt-for-a-github-app.md

### LINKED ISSUE #29925: Add pure shell script example
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/apps/creating-github-apps/authenticating-with-a-github-app/generating-a-json-web-token-jwt-for-a-github-app#example-using-python-to-generate-a-jwt

### What part(s) of the article would you like to see updated?

I would like a pure shell script example of how the JWT is supposed to be generated (that can be used inside a docker container). I don't want to depend of python or ruby just to generate a token and the examples don't work well inside a container. My latest attempt looks like this but for some reason the JWT is invalid:

```
#!/bin/bash

app_id=$1
pem=$2

header='{
	"typ": "JWT",
	"alg": "RS256",
	"iss": "$app_id"
}'

header=$(
	echo "${header}" | jq --arg time_str "$(date +%s)" \
	'
	($time_str | tonumber) as $time_num
	| .iat=$time_num
	| .exp=($time_num + 1)
	'
)

base64_encode()
{
	declare input=${1:-$(</dev/stdin)}
	printf '%s' "${input}" | base64 | tr -d '=' | tr '/+' '_-' | tr -d '\n'
}

json() {
	declare input=${1:-$(</dev/stdin)}
	printf '%s' "${input}" | jq -c .
}

hmacsha256_sign()
{
	declare input=${1:-$(</dev/stdin)}
	printf '%s' "${input}" | openssl dgst -binary -sha256 -sign $pem
}

header_base64=$(echo "${header}" | json | base64_encode)
signature=$(echo "${header_base64}" | hmacsha256_sign | base64_encode)

echo "${header_base64}.${signature}"
```

### Additional information

_No response_

### Content plan
Once a working shell script has been provided, update to include the new example

### PR DESCRIPTION
**Pull request by mark-mxwl**
### Why:

Closes: #29925 

### What's being changed (if available, include any code snippets, screenshots, or gifs):

Provided a working bash script for generating a JWT signed with RS256, as proposed in the comments. The script contains all of the required claims, and has been tested locally with the following sample usage and output:

Usage: 
```
./JWT5.sh 1234 BashPEM.pem
```
Output:
```
JWT: ewogICAgInR5cCI6IkpXVCIsCiAgICAiYWxnIjoiUlMyNTYiCn0.ewogICAgImlhdCI6MTcwMjI0MDIxNiwKICAgICJleHAiOjE3MDIyNDA4NzYsCiAgICAiaXNzIjoxMjM0Cn0.UnXejj2L8UOBtTkRIxqf7nqnQGRaMr-fx9GN7XIPo3yeLOYYNRNjCZDcQq0f9gqMwy1s1eZmBCFzDpdzlaTbxAGK865tSOX97KREx5XmC8pKBfYBYjWDCMQY2PBngSQtPU0ZN-OFi7CSS-Fec5GvejBZIfjAsCmC8XrKFSVyCEWEiFoIjfhGnsaZ-aQDBqsVgS42Sk0XPYF4Pm2EB0Yxt2rzsncgWRCGd9kj-EG5JbnJKm8DQtk5eHaLasJlHSDVVSSLc0-y_el02RxUkH2QiUsObZOftQAIzgFWrGvsM08mwvahPB6miaP_CiT5cDt3H1zuWTaDkzyUbqhMzZZialQqxuRkt8hNoZhdToKJp_mkNV1Qsss_pXWdCSDM0gvlWr7mU0bEiKczyWNR0f5jNZfZJbnknTfSttzcTcFbAXw2iWIxRNTRd_FqFVZYN3RXA9dAiq5l78ny5WLaWyLRyJjrIVioRY2F1pgxL1De4KlUn6NX1EbxzNNQjiGc7Yru
```

### Check off the following:

- [X] I have reviewed my changes in staging, available via the **View deployment** link in this PR's timeline.

  - For content changes, you will also see an automatically generated comment with links directly to pages you've modified. The comment won't appear if your PR only edits files in the `data` directory.
- [X] For content changes, I have completed the [self-review checklist](https://docs.github.com/en/contributing/collaborating-on-github-docs/self-review-checklist).

### CODE REVIEW

**Comment by nguyenalex836** [comment]
@mark-mxwl Thanks so much for opening a PR! I'll get this triaged for review ✨

**Comment by mark-mxwl** [comment]
Thx @nguyenalex836!

File: content/apps/creating-github-apps/authenticating-with-a-github-app/generating-a-json-web-token-jwt-for-a-github-app.md [hunk 1]
```diff
@@ -126,3 +126,52 @@ print(f"JWT:  {encoded_jwt}")
...
 
 This script will prompt you for the file path where your private key is stored and for the ID of your app. Alternatively, you can pass those values as inline arguments when you execute the script.
+
+### Example: Using Bash to generate a JWT
+
+{% note %}
+
+**Note:** You must pass your App ID and /path/to/PEM as arguments when running this script.
+
+{% endnote %}
```
Comment (skedwards88) [inline content/apps/creating-github-apps/authenticating-with-a-github-app/generating-a-json-web-token-jwt-for-a-github-app.md:136]: suggestion to write the full description
```suggestion
{% note %}

**Note:** You must pass your App ID and the file path where your private key is stored as arguments when running this script.

{% endnote %}
```

File: content/apps/creating-github-apps/authenticating-with-a-github-app/generating-a-json-web-token-jwt-for-a-github-app.md [hunk 2]
```diff
@@ -126,3 +126,52 @@ print(f"JWT:  {encoded_jwt}")
...
+
+```bash copy
+#!/usr/bin/env bash
+
+set -o pipefail
+
+app_id=$1 # App ID as first argument
+pem=$( cat $2 ) # /path/to/PEM as second argument
```
Comment (skedwards88) [inline content/apps/creating-github-apps/authenticating-with-a-github-app/generating-a-json-web-token-jwt-for-a-github-app.md:144]: same
```suggestion
pem=$( cat $2 ) # file path of the private key as second argument
```

File: content/apps/creating-github-apps/authenticating-with-a-github-app/generating-a-json-web-token-jwt-for-a-github-app.md [hunk 3]
```diff
@@ -126,3 +126,52 @@ print(f"JWT:  {encoded_jwt}")
...
+
+now=$(date +%s)
+iat=$((${now} - 60)) # Issues 60 seconds in the past.
+exp=$((${now} + 600)) # Expires 10 minutes in the future.
```
Comment (skedwards88) [inline content/apps/creating-github-apps/authenticating-with-a-github-app/generating-a-json-web-token-jwt-for-a-github-app.md:148]: nit suggestion for consistency with the earlier comments
```suggestion
iat=$((${now} - 60)) # Issues 60 seconds in the past
exp=$((${now} + 600)) # Expires 10 minutes in the future
```

**Review by skedwards88 — COMMENTED** [review]
Thanks for adding this example! I left some very minor suggestions that I will go ahead and incorporate, then we'll get this merged for you.

**Review by skedwards88 — APPROVED** [review]

**Comment by mark-mxwl** [comment]
Amazing, thx so much @skedwards88!! This was a fun one 🙌

**Pull request merged**